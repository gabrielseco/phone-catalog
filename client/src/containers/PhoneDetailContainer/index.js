export { default as PhoneDetailContainer } from './PhoneDetailContainer';
