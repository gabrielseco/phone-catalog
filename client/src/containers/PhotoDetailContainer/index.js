export { default as PhotoDetailContainer } from './PhotoDetailContainer';
