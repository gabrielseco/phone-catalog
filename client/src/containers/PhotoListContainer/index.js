export { default as PhotoListContainer } from './PhotoListContainer';
