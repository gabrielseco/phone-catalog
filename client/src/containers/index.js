export * from './Layout';
export * from './PhoneListContainer';
export * from './PhoneDetailContainer';
export * from './SpinnerContainer';
