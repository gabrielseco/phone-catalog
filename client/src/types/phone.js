// @flow
export type Phone = {
  id: number,
  name: string,
  manufacturer: string,
  price: string,
  image: string
};
