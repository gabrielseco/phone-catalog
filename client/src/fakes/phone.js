// @flow
import { type Phone } from './../types/phone';

export const phonesFake: Phone[] = [
  {
    id: 1,
    name: 'Iphone X',
    manufacturer: 'Apple Inc',
    price: '999$',
    image: 'https://img.shoppydoo.es/products/7/apple_iphone_x_64gb-300x300.jpg'
  },
  {
    id: 2,
    name: 'Samsung Galaxy S9',
    manufacturer: 'Samsung',
    price: '899$',
    image:
      'https://www.satugadget.com.my/content/images/thumbs/0008898_samsung-galaxy-s9-plus-256gb-purple-original-malaysia-set_300.jpeg'
  }
];
