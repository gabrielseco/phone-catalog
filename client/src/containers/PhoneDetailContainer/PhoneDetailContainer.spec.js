import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import {
  PhoneDetailContainerTesting,
  mapStateToProps
} from './PhoneDetailContainer';
import { phonesFake } from 'fakes/phone';
import { SpinnerContainer } from 'containers';

const dispatch = jest.fn();

describe('PhoneDetailContainer', () => {
  it('renders PhoneDetailContainer with a phone', () => {
    const wrapper = shallow(
      <PhoneDetailContainerTesting dispatch={dispatch} phone={phonesFake[0]} />
    );
    const tree = renderer
      .create(
        <PhoneDetailContainerTesting
          dispatch={dispatch}
          phone={phonesFake[0]}
        />
      )
      .toJSON();

    expect(wrapper).toBeDefined();
    expect(tree).toMatchSnapshot();
    expect(wrapper.find(SpinnerContainer).length).toBe(0);
    expect(dispatch).not.toHaveBeenCalled();
  });

  it('renders PhoneListContainer without the phone', () => {
    const wrapper = shallow(
      <PhoneDetailContainerTesting dispatch={dispatch} />
    );

    expect(wrapper).toBeDefined();
    expect(wrapper.find(SpinnerContainer).length).toBe(1);
    expect(dispatch).toHaveBeenCalled();
  });

  it('should test mapStateToProps', () => {
    const state = {
      phones: {
        phones: phonesFake,
        isLoading: false,
        isLoaded: false
      }
    };

    const ownProps = {
      match: {
        params: {
          id: phonesFake[0].id.toString()
        }
      }
    };
    expect(mapStateToProps(state, ownProps)).toEqual({
      phone: phonesFake[0]
    });
  });
});
