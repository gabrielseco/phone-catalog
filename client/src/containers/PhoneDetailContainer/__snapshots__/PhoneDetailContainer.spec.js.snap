// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneDetailContainer renders PhoneDetailContainer with a phone 1`] = `
<div
  className="container"
>
  <div
    className="imageContainer"
  >
    <img
      alt="Iphone X"
      src="https://img.shoppydoo.es/products/7/apple_iphone_x_64gb-300x300.jpg"
    />
  </div>
  <div
    className="info"
  >
    <h2
      className="title"
    >
      Iphone X
    </h2>
    <p
      className="manufacturer"
    >
      Built by: 
      Apple Inc
    </p>
    <p
      className="price"
    >
      999$
    </p>
    <p
      className="description"
    >
      Lorem fistrum diodeno nostrud te voy a borrar el cerito ut te va a hasé pupitaa. Officia a wan aute irure tiene musho peligro sed nostrud officia qué dise usteer. Jarl labore exercitation reprehenderit consectetur ex. Qué dise usteer irure aliquip officia esse apetecan et ese pedazo de. Ese que llega ut fistro consectetur pupita jarl caballo blanco caballo negroorl reprehenderit. Ut commodo elit nostrud de la pradera se calle ustée. Torpedo ex magna nisi elit.
    </p>
  </div>
</div>
`;
